src/main/java/InputHandler.java

```diff
@@ -0,0 +1,40 @@
+public class InputHandler {
+
+    public static void processInputCases(String input) {
+        if (input.equals("list")) {
+            TaskList.listTasks();
+            return;
+        }
+
+        String[] words = input.split(" ");
+        int taskIndex;
+
+        switch (words[0]) {
+        case "mark":
+            taskIndex = Integer.parseInt(words[1]) - 1;
+            TaskList.markTask(taskIndex);
+            break;
+        case "unmark":
+            taskIndex = Integer.parseInt(input.substring(7)) - 1;
+            TaskList.unmarkTask(taskIndex);
+            break;
+        case "todo":
+            TaskList.addTask(new ToDo(false, input.substring(5)));
+            break;
+        case "deadline":
+            String[] deadlineParts = input.substring(9).split("/by", 2);
+            TaskList.addTask(new Deadline(false, deadlineParts[0], deadlineParts[1]));
+            break;
```

Comment:
I like the clear logic, use of switch and how compact everything is! One thing to consider is adding more error handling, especially around Integer.parseInt() and array indexing, to prevent crashes on invalid inputs. It might also be worth refactoring the splitting of strings into helper methods for better readability .